extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java

```diff
@@ -444,6 +444,14 @@ private void addProvider(MethodCreator ctor, AssignableResultHandle target, Inde
                 MethodDescriptor.ofMethod(BeanGrabber.class, "getBeanIfDefined", Object.class, Class.class),
                 ctor.loadClass(providerClass));
 
+        ClassInfo providerClassInfo;
+        if (registerProvider.target() != null) {
+            providerClassInfo = registerProvider.target().asClass();
```

Comment:
I don't understand this part, this line seems to be setting providerClassInfo to the jaxrs-annotated interface and line 452 seems to be setting it to the provider class.

Reply:
Both branches get the `ClassInfo` of the class registered with `@RegisterProvider(SomeClass.class)`. 
The first branch gets it for the case where a single `@RegisterProvider` annotation is present on the interface, the second branch is used for the case of repeatable annotations

Reply:
`registerProvider.target()` returns the class on which the registerProvider annotation was placed, doesn't it?

Reply:
OK right, my dumb mistake. Thanks for spotting it

Reply:
Fixed